File: PennMobile/src/main/java/com/pennapps/labs/pennmobile/GsrBuildingAdapter.java
package com.pennapps.labs.pennmobile;

import android.content.Context;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.pennapps.labs.pennmobile.classes.GSR;
import com.pennapps.labs.pennmobile.classes.GSRContainer;
import com.pennapps.labs.pennmobile.classes.GSRContainerSlot;
import com.pennapps.labs.pennmobile.classes.GSRSlot;

import java.util.ArrayList;

/**
 * Created by Andrew on 11/5/2017.
 * RecylerView Adapter for a specific building with GSRs.
 */

public class GsrBuildingAdapter extends RecyclerView.Adapter<GsrBuildingHolder> {

    Context context;
    ArrayList<GSRContainer> gsrs;
    String gsrLocationCode;

    public GsrBuildingAdapter(Context context, ArrayList<GSRContainer> _gsrs, String _gsrLocationCode) {
        this.context = context;
        this.gsrs = _gsrs;
        this.gsrLocationCode = _gsrLocationCode;
    }

    @Override
    public GsrBuildingHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.gsr_building, parent, false);
        return new GsrBuildingHolder(view);
    }

    @Override
    public void onBindViewHolder(GsrBuildingHolder holder, int position) {
        if (position < getItemCount()) {
            RecyclerView gsrRoomsRecyclerView = holder.recyclerView;
            if (gsrRoomsRecyclerView != null) {
                LinearLayoutManager gsrRoomsLayoutManager = new LinearLayoutManager(context,
                        LinearLayoutManager.HORIZONTAL, false);
                gsrRoomsRecyclerView.setLayoutManager(gsrRoomsLayoutManager);

                //now define arrays
                ArrayList<String> times = new ArrayList<String>();
                ArrayList<String> dates = new ArrayList<String>();
                ArrayList<String> ids = new ArrayList<String>();

                for (int j = 0; j < gsrs.get(position).getAvailableGSRSlots().size(); j++)
                {
                    GSRContainerSlot gsrslot = gsrs.get(position).getAvailableGSRSlots().get(j);
                    times.add(gsrslot.getTimeRange());
                    dates.add(gsrslot.getDateNum());
                    ids.add(gsrslot.getElementId());
                }

                gsrRoomsRecyclerView.setAdapter(new GsrRoomAdapter(times, ids, gsrLocationCode, context, dates));
                holder.gsrBuildingName.setText(gsrs.get(position).getGsrName());
            }
        }
    }

    @Override
    public int getItemCount() {
        return gsrs.size();
    }
}

File: PennMobile/src/main/java/com/pennapps/labs/pennmobile/GsrBuildingAdapter.kt
package com.pennapps.labs.pennmobile

import android.content.Context
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.ViewGroup
import com.pennapps.labs.pennmobile.classes.GSRContainer
import org.joda.time.DateTime
import java.util.*

class GsrBuildingAdapter(internal var context: Context, internal var gsrs: ArrayList<GSRContainer>,
                         internal var gsrLocationCode: String) : RecyclerView.Adapter<GsrBuildingHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): GsrBuildingHolder {
        val view = LayoutInflater.from(parent.context)
                .inflate(R.layout.gsr_building, parent, false)
        return GsrBuildingHolder(view)
    }

    override fun onBindViewHolder(holder: GsrBuildingHolder, position: Int) {
        if (position < itemCount) {
            val gsrRoomsRecyclerView = holder.recyclerView
            if (gsrRoomsRecyclerView != null) {
                val gsrRoomsLayoutManager = LinearLayoutManager(context,
                        LinearLayoutManager.HORIZONTAL, false)
                gsrRoomsRecyclerView.layoutManager = gsrRoomsLayoutManager

                //now define arrays
                val timeRanges = ArrayList<String>()
                val startTimes = ArrayList<DateTime>()
                val ids = ArrayList<String>()

                for (j in 0 until gsrs[position].availableGSRSlots.size) {
                    val gsrslot = gsrs[position].availableGSRSlots[j]
                    timeRanges.add(gsrslot.timeRange)
                    startTimes.add(gsrslot.startTime)
                    ids.add(gsrslot.elementId)
                }
                gsrRoomsRecyclerView.adapter = GsrRoomAdapter(timeRanges, ids, gsrLocationCode, context, startTimes)
                holder.gsrBuildingName?.text = gsrs[position].gsrName
            }
        }
    }

    override fun getItemCount(): Int { return gsrs.size }
}